File: 75-sort-colors/75-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        
        int red = 0, white =0, blue =0;
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i]==0)
            {
              red++;  
            }
            else if(nums[i]==1)
            {
                white++;
            }
            else if(nums[i]==2)
            {
              blue++;  
            }
        }
        vector<int> v;
        vector<int> v1(red,0);
        v.insert(v.end(), v1.begin(), v1.end());
        vector<int> v2(white,1);
            v.insert(v.end(), v2.begin(), v2.end());
        vector<int> v3(blue,2);
            v.insert(v.end(), v3.begin(), v3.end());
        nums = v;
            
    }
};

File: 75-sort-colors/75-sort-colors.java
class Solution {
    
    public void swap(int[] arr, int i,int j){
        int temp = arr[i];
        arr[i]=arr[j];
        arr[j]=temp;
    }
    public void sortColors(int[] nums) {
        
       
      
    int p0 = 0, curr = 0;
    // for all idx > k : nums[idx > k] = 2
    int p2 = nums.length - 1;

    int tmp;
    while (curr <= p2) {
      if (nums[curr] == 0) {
        // swap p0-th and curr-th elements
        // i++ and j++
        swap(nums,curr++,p0++);
      }
      else if (nums[curr] == 2) {
        // swap k-th and curr-th elements
        // p2--
       swap(nums,curr,p2--);
      }
      else curr++;
    }
        
    }
}